cfn-resources/project/cmd/resource/resource.go

```diff
@@ -470,9 +462,9 @@ func readProjectSettings(client *mongodbatlas.Client, id string, currentModel *M
 	}
 
 	// Set api-keys
-	apiKeys := readKeys(*currentModel.Id, projectAPIKeys, currentModel)
+	// apiKeys := readKeys(*currentModel.Id, projectAPIKeys, currentModel)
```

Comment:
👀 Commented code

Reply:
This was a  big story @colm-quinn :)
An extra api key(default) gets added  to project when we return the values to projectapikeys[] and then cfn test fails because the input array value doesn't match with the return value. So we have introduced hack to pass the test cases. But now I am thinking, anyhow apikeys are provided from inputs and these are not something created by server hence tried to avoid returning the apikeys back.

Reply:
Yes, I'd expect the API keys only come from reading the input profile, so it won't be read from the current model. It looks like the apikeys can be removed from the model and instead always fetched using the secrets manager.

Also, the commented line can be removed, as it adds no value.

Wdyt?

Reply:
@colm-quinn 
**ProjectApiKeys**  are associated to **Project** and can have different role.
 **ProjectApiKeys** are different from **ApiKeys** that we use in our model  which are used to Connect to Atlas APIs. 

We followed the way terraform implemented the project Resource. Project creation involves multiple APIs including adding teams[], ApiKeys[]  and settings to project.

In this case, User can use secret manager to add keys to a project but we don't want to return in the output. 

Hope you got it!

Reply:
Ahh thank you - I missed it was the project keys from the API call. That clears everything up. Thanks!